tracer/src/Datadog.Trace/Propagators/W3CTraceContextPropagator.cs

```diff
@@ -90,25 +89,30 @@ internal static string CreateTraceParentHeader(SpanContext context)
 
         internal static string CreateTraceStateHeader(SpanContext context)
         {
-            var samplingPriority = SamplingPriorityToString(context.TraceContext?.SamplingPriority ?? context.SamplingPriority);
-            StringBuilder? sb = null;
+            var sb = StringBuilderCache.Acquire(100);
 
             try
             {
-                sb = StringBuilderCache.Acquire(100);
                 sb.Append("dd=");
 
+                // sampling priority ("s:<value>")
+                var samplingPriority = SamplingPriorityToString(context.TraceContext?.SamplingPriority);
```

Comment:
Heh, yeah that's a weird one. Historically we have used this in many places "just in case" for some reason the `TraceContext` doesn't have a sampling priority and the `SpanContext` does, but in practice this should never happen, so I wanted to remove the unnecessary check. I'm hoping they will all go away soon anyway if I get to do my `SpanContext` cleanup 😉